cmake/Conan.cmake

```diff
@@ -1,33 +1,38 @@
 macro(run_conan)
   # Download automatically, you can also just copy the conan.cmake file
   if(NOT EXISTS "${CMAKE_BINARY_DIR}/conan.cmake")
-    message(STATUS "Downloading conan.cmake from https://github.com/conan-io/cmake-conan")
-    file(DOWNLOAD "https://github.com/conan-io/cmake-conan/raw/v0.16.1/conan.cmake" "${CMAKE_BINARY_DIR}/conan.cmake")
+    message(
+      STATUS
+        "Downloading conan.cmake from https://github.com/conan-io/cmake-conan")
+    file(
+      DOWNLOAD
+      "https://raw.githubusercontent.com/conan-io/cmake-conan/v0.16.1/conan.cmake"
+      "${CMAKE_BINARY_DIR}/conan.cmake"
+      EXPECTED_HASH
+        SHA256=396e16d0f5eabdc6a14afddbcfff62a54a7ee75c6da23f32f7a31bc85db23484
+      TLS_VERIFY ON)
   endif()
 
-  set (ENV{CONAN_REVISIONS_ENABLED} 1)
+  set(ENV{CONAN_REVISIONS_ENABLED} 1)
+  list(APPEND CMAKE_MODULE_PATH ${CMAKE_BINARY_DIR})
+  list(APPEND CMAKE_PREFIX_PATH ${CMAKE_BINARY_DIR})
+
   include(${CMAKE_BINARY_DIR}/conan.cmake)
 
-  conan_add_remote(
-    NAME
-    conan-center
-    URL
-    https://api.bintray.com/conan/conan/conan-center)
+  # conan_add_remote(NAME conan-center URL https://conan.bintray.com)
+  conan_add_remote(NAME cci URL https://center.conan.io INDEX 0)
 
-  conan_add_remote(
-    NAME
-    bincrafters
-    URL
-    https://bincrafters.jfrog.io/artifactory/api/conan/public-conan)
+  conan_cmake_autodetect(settings)
 
-  conan_cmake_run(
-    CONANFILE
-    conanfile.py
+  conan_cmake_install(
```

Comment:
I would like to add comments that make it abundantly clear what is happening here, and that `conan_cmake_install` is automatically detecting `conanfile.py` and running it. The cmake-conan readme makes no mention of this; I had to go digging in the issues at https://github.com/conan-io/cmake-conan/issues/331 to find this out. I'm sure that anyone reading this code alongside the cmake-conan readme will wonder why we skipped the call to `conan_cmake_configure`, which the readme recommends doing. 

For the record, calling `conan_cmake_configure` autogenerates a Conanfile.txt; there's no need to do that if you already have a Conanfile.py.

Reply:
Right, that's a good point. I considered adding a comment for the directory used in conan_cmake_install. I'll do that now.

Reply:
Much improved!